File: js/027_RemoveElement.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    var tail = nums.length - 1;
    for( var i = 0; i <= tail; ++i){
        if(nums[i] === val){
            nums[i] = nums[tail];
            --i;
            --tail;
        }
    }
    return tail + 1;
};

File: cpp/027_RemoveElement.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int tail = nums.size()-1;
        for( int i = 0; i <= tail; ++i){
            if(nums[i] == val){
                nums[i] = nums[tail];
                tail--;
                i--;
            }
        }
        return tail+1;
    }
};